internal/controllers/eventsource/controller.go

```diff
@@ -0,0 +1,100 @@
+package eventsource
+
+import (
+	"context"
+	"fmt"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	eshandler "github.com/kanopy-platform/argoslower/internal/admission/eventsource"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	esv1alpha1 "github.com/argoproj/argo-events/pkg/apis/eventsource/v1alpha1"
+	eslister "github.com/argoproj/argo-events/pkg/client/eventsource/listers/eventsource/v1alpha1"
+)
+
+type EventSourceIngressController struct {
+	esLister eslister.EventSourceLister
+	//	config    EventSourceIngressControllerConfig
+}
+
+func NewEventSourceIngressController(esl eslister.EventSourceLister) *EventSourceIngressController {
+	return &EventSourceIngressController{
+		esLister: esl,
+	}
+}
+
+func (e *EventSourceIngressController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	eventSource, err := e.esLister.EventSources(req.NamespacedName.Namespace).Get(req.NamespacedName.Name)
+	if err != nil && !k8serror.IsNotFound(err) {
+		log.Error(err, fmt.Sprintf("unable to get eventsource %v", req))
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	if err = e.reconcile(ctx, eventSource.DeepCopy(), req.NamespacedName); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	return ctrl.Result{}, nil
+
+}
+
+func (e *EventSourceIngressController) reconcile(ctx context.Context, es *esv1alpha1.EventSource, nsn types.NamespacedName) error {
+	log := log.FromContext(ctx)
+	if es == nil {
+		//TODO: implement garbage collection we should be able to use the NamespacedName to associated vanity resources to the namespace/eventsource
+		// event source svc labels
+		//   labels:
+		//      controller: eventsource-controller
+		//      eventsource-name: demo-day
+		//      owner-name: demo-day
+		return nil
+	}
+
+	hookType, ok := es.Annotations[eshandler.DefaultAnnotationKey]
+	if !ok {
+		return nil
+	}
+
+	switch hookType {
+	case "github":
+		log.Info("populate ingress config for github")
+
+	case "jira":
+		log.Info("populate ingress config for a generic webhook")
+
+		if es.Spec.Webhook != nil {
```

Comment:
Just a note for future PRs - what do we consider a "generic" webhook?

Reply:
EventSources have a number of different Specs fields that control what kind of event source it is. Github is its own first class field that functionally works like a webhook with special sauce.

The rest of the sources we are looking to support will all be of the generic Webhook type EventSource. The annotation we are parsing will tell us which source service the webhook is associated with, e.g. jira, which will drive which ip getter we attach for generating our AuthorizationPolicy.

Reply:
Right, forgot "generic" was a term in argo events.